### PR TITLE
2904 20 creating styles no longer attaches it to themes

### LINKED ISSUE #2904: 2.0: Creating styles no longer attaches it to themes
We seem to have caused a regression. Creating styles in 2.0 creates styles as expected, but it does not automatically attach them to the theme.

Production did that.

[todo: add recording]

### PR DESCRIPTION
**Pull request by robinhoodie0823**
### Why does this PR exist?
Current plugin has problem in token resolving logic, so it merges same name tokens but with different internal parent.

Closes #2904  

### What does this pull request do?
Updated token resolving logic to check token's internal parent to prevent merging of same name tokens.

### Additional Notes (if any)

https://github.com/tokens-studio/figma-plugin/assets/103296157/41d34ed6-dc13-4833-a786-54e31048c931

### CODE REVIEW

**Review by cuserox — APPROVED** [review]
Looks like it works well. Tested locally 👌

**Review by six7 — CHANGES_REQUESTED** [review]
The PR has a flaw:

You selected "Prefix styles with active theme name", but the styles that were created are only prefixed for the `light` theme - the `dark` tokens are not prefixed at all?

CleanShot 2024-07-05 at 12 29 25@2x

In that case the expectation is that there's 2 groups of styles created:

`light/...`
`dark/...`

File: packages/tokens-studio-for-figma/src/utils/TokenResolver.ts [hunk 1]
```diff
@@ -114,7 +114,7 @@ class TokenResolver {
     const memoKey = token.name || undefined;
 
     // If we have a cache hit, we can return it
-    if (memoKey && this.memo.has(memoKey)) {
+    if (memoKey && this.memo.has(memoKey) && token.internal__Parent === this.memo.get(memoKey)?.internal__Parent) {
```
[the remark this answers is not in the document]
Comment (six7) [reply packages/tokens-studio-for-figma/src/utils/TokenResolver.ts:117]: For me right now it's hard to understand why we do this and I'd like to understand.

[the remark this answers is not in the document]
Comment (robinhoodie0823) [reply packages/tokens-studio-for-figma/src/utils/TokenResolver.ts:117]: Yes. The current token resolving logic does not check token's parent name, so it merges tokens with same name but in different token sets.

**Review by six7 — APPROVED** [review]

**Pull request merged**